pkg/asset/installconfig/azure/client.go

```diff
@@ -310,6 +310,14 @@ func (c *Client) GetHyperVGenerationVersion(ctx context.Context, instanceType st
 				g = strings.ToUpper(g)
 				generations.Insert(g)
 			}
+			// If a VM image has been supplied, we need to pick a HyperVGen
+			// that's compatible with that of the image or fail otherwise
+			if imageHyperVGen != "" {
+				if generations.Has(imageHyperVGen) {
+					return imageHyperVGen, nil
+				}
+				return "", fmt.Errorf("instance type %s supports HyperVGenerations %v but the specified image is for HyperVGeneration %s; to correct this issue either specify a compatible instance type or change the HyperVGeneration for the image by using a different SKU", instanceType, generations.UnsortedList(), imageHyperVGen)
+			}
```

Comment:
I missed this when doing my previous review: I don't think it's great to be doing the validation inside this function. Typically these client functions just return data and then we validate using that, so I think doing the validation inside is a little confusing.

I get why you're doing it, this function doesn't return all of the Generations, because it defaults to V2 if present. Aren't the hypervgenerations included in the capabilities map? Could we use that instead?

Reply:
WDYT about 7d0707f004b88173613648eadc94c044bf30ca20 ?